geoips/plugins/modules/procflows/single_source.py

```diff
@@ -1810,39 +1888,79 @@ def call(fnames, command_line_args=None):
         and not resampled_read
     ):
         pid_track.print_mem_usg()
-        LOG.interactive(
-            "Reading full dataset " "with reader '%s'...", reader_plugin.name
-        )
+        LOG.interactive("Reading full dataset with reader '%s'...", reader_plugin.name)
         xobjs = reader_plugin(
             fnames, metadata_only=False, chans=variables, **reader_kwargs
         )
 
     pid_track.print_mem_usg()
-    # If we have a product of type "unsectored_xarray_dict_to_output_format"
-    # process it here
-    # This will not have any required area_defs
-    if prod_plugin.family == "unsectored_xarray_dict_to_output_format":
+    # If we have a product of a family that does not require an area definition,
+    # and operates on dictionaries of xarrays, process it here.
+    # This will not have any required area_defs, so will never make it into
+    # the loop through all area_defs below.
+    if prod_plugin.family in PRODUCT_FAMILIES_OF_XARRAY_DICT_WITHOUT_AREA:
         LOG.interactive(
-            "Reading full dataset " "for unsectored products " "with reader '%s'...",
+            "Reading full dataset for unsectored products with reader '%s'...",
             reader_plugin.name,
         )
         xdict = reader_plugin(fnames, metadata_only=False, **reader_kwargs)
+        # If this product family requires an algorithm to be applied, apply here.
```

Comment:
This update was required to fix the smap unsectored test - it was getting passed to get alg xarray, which failed